### PR TITLE
feat(nodebuilder/blob/cmd): Allow submit multiple blobs through file input

### PR DESCRIPTION
**Pull request by sontrinh16**
## Description

Closed: #2552 

## Testing:
testing with `mocha` testnet and get the following result ( Note: the `commitments` field contains an array of `commitments` from all the blobs):

Screenshot 2023-12-14 at 14 55 00

where `test.json` file contain:
`{
			"Blobs": [
				{
					"namespace": "0x00010203040506070809",
					"blobData": "0x676d"
				},
				{
					"namespace": "0x42690c204d39600fddd3",
					"blobData": "0x676d"
				}
			]
		}`

### CODE REVIEW

**Comment by Wondertan** [comment]
@vgonkivs, please take a look

**Review by vgonkivs — CHANGES_REQUESTED** [review]
Thanks for your PR @sontrinh16. Left some comments

File: nodebuilder/blob/cmd/blob.go [hunk 1]
```diff
@@ -18,6 +20,10 @@ var (
 
 	fee      int64
 	gasLimit uint64
+
+	// FlagFileInput allows the user to provide file path to the json file
```
Comment (vgonkivs) [inline nodebuilder/blob/cmd/blob.go:24]:
```suggestion
	// flagFileInput allows the user to provide file path to the json file
```

File: nodebuilder/blob/cmd/blob.go [hunk 2]
```diff
@@ -130,11 +138,46 @@ var getAllCmd = &cobra.Command{
...
+		if path != "" {
+			if filepath.Ext(path) != ".json" {
+				return fmt.Errorf("invalid file extension, require json got %s", filepath.Ext(path))
+			}
+
+			return nil
+		}
+
+		if len(args) < 2 {
+			return errors.New("PayForBlobs requires two arguments: namespace and blobData")
```
Comment (vgonkivs) [inline nodebuilder/blob/cmd/blob.go:158]:
```suggestion
			return errors.New("submit requires two arguments: namespace and blobData")
```

File: nodebuilder/blob/cmd/blob.go [hunk 3]
```diff
@@ -143,33 +186,66 @@ var submitCmd = &cobra.Command{
...
+				return err
+			}
+
+			jsonBlobs = append(jsonBlobs, paresdBlobs...)
+		} else {
+			jsonBlobs = append(jsonBlobs, blobJSON{Namespace: args[0], BlobData: args[1]})
+		}
+
+		var blobs []*blob.Blob
+		var conmmitments []blob.Commitment
```
Comment (vgonkivs) [inline nodebuilder/blob/cmd/blob.go:208]: typo

**Review by vgonkivs — COMMENTED** [review]
Left some nits but other than that great work 🚀 . Thank you. 

PS. tested locally. Works for me
Снимок экрана 2024-01-09 в 16 45 20

File: nodebuilder/blob/cmd/blob.go (hunk 1 above)
Comment (sontrinh16) [reply nodebuilder/blob/cmd/blob.go]: yikes miss this one

**Comment by jcstein** [comment]
testing on mocha-4 testnet using 2 identical blobs

test.json
```json
{ "Blobs": [ { "namespace": "0x0000000000004a6f7368", "blobData": "0x676d" }, { "namespace": "0x0000000000004a6f7368", "blobData": "0x676d" } ] }
```

```bash
$ celestia blob submit --input-file test.json --node.store $NODE_STORE
{
  "result": {
    "height": 910885,
    "commitments": [
      "uYcW7vh+XB1VCbWSjTQJFA3PV89/089LyhaNQ5hXxjY=",
      "uYcW7vh+XB1VCbWSjTQJFA3PV89/089LyhaNQ5hXxjY="
    ]
  }
}
```

**Review by distractedm1nd — DISMISSED** [review]
utACK

**Review by renaynay — DISMISSED** [review]
utack

**Review by vgonkivs — APPROVED** [review]

**Review by ramin — APPROVED** [review]

**Comment by ramin** [comment]
@sontrinh16 can you rebase main back onto this please? Then it'll be ready to merge

**Comment by ramin** [comment]
@sontrinh16 can you rebase main one more time, or if you toggle the "let admin make changes to PR" we can do this and get it merged, thanks

**Comment by sontrinh16** [comment]
> @sontrinh16 can you rebase main one more time, or if you toggle the "let admin make changes to PR" we can do this and get it merged, thanks

hmmm i can't find where to toggle but i can add you guys as collaborators

**Review by walldiss — APPROVED** [review]
Thank you for contribution! Looks good 👍

**Comment by jcstein** [comment]
I have not tested this on the current version of the PR, just at https://github.com/celestiaorg/celestia-node/pull/3008/commits/7587f80696b063902c84e52875efb66492d8c8d0

**Comment by sontrinh16** [comment]
i have tested with the newest commit 

image

**Pull request merged**